performance_tests/locustfile.py

```diff
@@ -0,0 +1,106 @@
+import secrets
+
+from locust import HttpUser, between, task
+
+CERT_PATH = False
+AYR_AAU_USER_USERNAME = ""
+AYR_AAU_USER_PASSWORD = ""
+KEYCLOAK_AUTH_URL = ""
+KEYCLOAK_CLIENT_ID = ""
+KEYCLOAK_CLIENT_SECRET = ""
+
+
+class User(HttpUser):
+    # Set the wait time between task executions (e.g., between 1 and 5 seconds)
+    wait_time = between(1, 5)
+
+    # Define tasks
+    @task
+    def index(self):
+        self.client.get("/", verify=CERT_PATH)
+
+    @task
+    def accessibility(self):
+        self.client.get("/accessibility", verify=CERT_PATH)
+
+    @task
+    def terms_of_use(self):
+        self.client.get("/terms-of-use", verify=CERT_PATH)
+
+    @task
+    def how_to_use(self):
+        self.client.get("/how-to-use-this-service", verify=CERT_PATH)
+
+    @task
+    def privacy(self):
+        self.client.get("/privacy", verify=CERT_PATH)
+
+    @task
+    def cookies(self):
+        self.client.get("/cookies", verify=CERT_PATH)
+
+
+class KeycloakUser(HttpUser):
+    wait_time = between(1, 5)
+
+    def on_start(self):
+        self.token, self.refresh_token = self.authenticate()
+
+    def authenticate(self):
+        token_url = KEYCLOAK_AUTH_URL
+        client_id = KEYCLOAK_CLIENT_ID
+        client_secret = KEYCLOAK_CLIENT_SECRET
+        username = AYR_AAU_USER_USERNAME
```

Comment:
ideally we should create username/password on the fly similarly to the e2e tests and just use those credentials in order to get the token via password grant type, otherwise would we need a specific test account or our developer accounts not to have 2fa enabled?

(see e2e_tests/conftest.py line 91)

Reply:
Yes this does require a specific account with 2FA disabled. There is a test user which could be kept specifically for test purposes or created on the fly without 2FA.